great_expectations_cloud/agent/agent.py

```diff
@@ -209,14 +208,24 @@ def _handle_event_as_thread_exit(
 
         # get results or errors from the thread
         error = future.exception()
+        # TODO Clean up ifs
         if error is None:
             result: ActionResult = future.result()
-            status = JobCompleted(
-                success=True,
-                created_resources=result.created_resources,
-            )
-            print(f"Completed job: {event_context.event.type} ({event_context.correlation_id})")
+
+            # TODO This result may have an UnknownEvent error
+            if result.event is UnknownEvent:
```

Comment:
this is the crux of changes

Reply:
Previously an unknown event would be re-enqueued so that it potentially could be processed by a different agent that does know how to handle it.
This looks like it would break flow.